### ISSUE TITLE
PR build limitation: test framework takes the default extension repo branch instead of current change

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#17595]
Copied from 
* https://github.com/ibmruntimes/openj9-openjdk-jdk/issues/506

`extended.openjdk` PR build still shows the errors https://github.com/eclipse-openj9/openj9/issues/16074 due to the test framework taking the default JDK19 extension repo instead of this PR.
Launched [an internal grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/28476/consoleFull) w/ this change and the three tests in question are skipped or passed.

_Originally posted by @JasonFengJ9 in https://github.com/ibmruntimes/openj9-openjdk-jdk19/issues/34#issuecomment-1272533659_

Related issues
* https://github.com/ibmruntimes/openj9-openjdk-jdk/issues/613

[Internal issue - OpenJ9 pipelines should pass openjdk repo/branch to test jobs](https://github.ibm.com/runtimes/infrastructure/issues/7363)

**Comment by pshipton** [comment]
Looking at a grinder, the parameters for test are JDK_REPO and JDK_BRANCH. At the top level job (https://openj9-jenkins.osuosl.org/job/Pipeline_Build_Test_JDK11_aarch64_linux/1475/console) we see
`09:39:38  Using OPENJDK_REPO = https://github.com/ibmruntimes/openj9-openjdk-jdk11.git OPENJDK_BRANCH = openj9`
In the test job (https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.openjdk_aarch64_linux_Personal/20/consoleFull), there is no mention of these parameters.

@AdamBrousseau said:

So the ask is that any PR build that changes code in extensions, should pass JDK_REPO & JDK_BRANCH to the test jobs. Correct? I'm not sure I was thinking of the same known issue originally.
Does the test code have ability at the moment to fetch PR refs and checkout a merge ref of extensions repos @llxia?

**Comment by llxia** [comment]
> So the ask is that any PR build that changes code in extensions, should pass JDK_REPO & JDK_BRANCH to the test jobs. Correct? 

Yes, we need set JDK_REPO and JDK_BRANCH for the extensions repos in the test job. 
Not sure what was the reason for setting the names differently in the job parameters.

> Does the test code have ability at the moment to fetch PR refs and checkout a merge ref of extensions repos

The test job takes JDK_REPO, JDK_BRANCH and OPENJDK_SHA. If the correct ref is passed in, the test job is able to get the extension repo materials.

![image](https://github.com/eclipse-openj9/openj9/assets/19273206/8a16813b-32b2-4235-9b68-68ba7214ad25)
